plugins/by-name/lint/default.nix

```diff
@@ -225,43 +172,81 @@ in
           callback = defaultCallback;
         };
       };
-  };
 
-  config = mkIf cfg.enable {
-    extraPlugins = [ cfg.package ];
+    customLinters = defaultNullOpts.mkAttrsOf (with types; either str (mkLinterOpts true)) { } ''
+      Configure the linters you want to run per file type.
+      It can be both an attrs or a string containing the lua code that returns the appropriate
+      table.
+    '';
+
+    linters =
+      defaultNullOpts.mkAttrsOf (mkLinterOpts false)
+        {
+          phpcs.args = [
+            "-q"
+            "--report=json"
+            "-"
+          ];
+        }
+        ''
+          Configure the linters you want to run.
+          You can also add custom linters here.
+        '';
+
+    lintersByFt =
+      defaultNullOpts.mkAttrsOf (types.listOf types.str)
+        {
+          text = [ "vale" ];
+          json = [ "jsonlint" ];
+          markdown = [ "vale" ];
+          rst = [ "vale" ];
+          ruby = [ "ruby" ];
+          janet = [ "janet" ];
+          inko = [ "inko" ];
+          clojure = [ "clj-kondo" ];
+          dockerfile = [ "hadolint" ];
+          terraform = [ "tflint" ];
+        }
+        ''
+          Configure the linters you want to run per file type.
+        '';
+  };
 
-    extraConfigLua =
+  extraConfig = cfg: {
+    autoCmd = lib.optionals (cfg.autoCmd != null) [ cfg.autoCmd ];
+    plugins.lint.luaConfig.content =
       ''
-        __lint = require('lint')
-        __lint.linters_by_ft = ${lib.nixvim.toLuaObject cfg.lintersByFt}
+        local lint = require('lint')
```

Comment:
Only concern is changing the naming here. I believe we've been doing __ prefixes to prevent messing with a user's config.

Reply:
Reverted back to `__lint`.